modules/Bio/EnsEMBL/Compara/RunnableDB/PairAligner/ChunkAndGroupDna.pm

```diff
@@ -128,33 +129,48 @@ sub create_chunks {
             my ($coord_system_name, $region_name, $region_start, $region_end) = split(/:/, $region);
             my $region_dnafrag = $dnafrag_dba->fetch_by_GenomeDB_and_name($genome_db, $region_name);
             die "Unknown dnafrag region '$region_name'\n" unless $region_dnafrag;
-            $region_dnafrag->{_start} = $region_start if $region_start;
-            $region_dnafrag->{_end} = $region_end if $region_end;
-            $dnafrags->{"$coord_system_name:$region_name"} = $region_dnafrag;
+            my $locus = bless {
+                'dnafrag'         => $region_dnafrag,
+                'dnafrag_start'   => $region_start || 1,
+                'dnafrag_end'     => $region_end || $region_dnafrag->length,
+                'dnafrag_strand'  => 1,
+            }, 'Bio::EnsEMBL::Compara::Locus';
+            push @regions_to_align, $locus;
         }
     } else {
         my $dnafrag_list = $dnafrag_dba->fetch_all_by_GenomeDB(
             $genome_db,
             -IS_REFERENCE       => $self->param('include_non_reference') ? undef : 1,
             -CELLULAR_COMPONENT => $self->param('only_cellular_component'),
         );
-        $dnafrags = { map {$_->coord_system_name . ':' . $_->name => $_} @$dnafrag_list };
+
+        # Preload all the alt-regions at once. Not using Utils::Preloader
+        # because the latter assumes that alt-regions (Locus objects) would
+        # have a dbID() method, and also does not create fields linked to undef
```

Comment:
This is a reference to how `Utils::Preloader::_load_and_attach_all` works, esp. [line 184](https://github.com/Ensembl/ensembl-compara/blob/release/101/modules/Bio/EnsEMBL/Compara/Utils/Preloader.pm#L184). But in fact I'll be adding a method in `Utils::Preloader` to load all alt-regions in the next pull-request because the functionality is actually needed somewhere else too, but it won't be based on `_load_and_attach_all`.